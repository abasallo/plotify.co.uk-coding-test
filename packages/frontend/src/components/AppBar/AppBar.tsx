import React from 'react'
import { useNavigate } from 'react-router-dom'

import logo from '../../images/logo.png'

import constants from '../../modules/constants'

import { Logo, MaterialUIStyledAppBar, Toolbar } from './AppBar.styled.components'

const AppBar = () => {
  const navigate = useNavigate()

  const handleLogoClick = () => navigate(constants.ROUTE_URL_ROOT)

  return (
    <MaterialUIStyledAppBar position="static">
      <Toolbar>
        <Logo src={logo} onClick={() => handleLogoClick()} />
      </Toolbar>
    </MaterialUIStyledAppBar>
  )
}

export default AppBar
