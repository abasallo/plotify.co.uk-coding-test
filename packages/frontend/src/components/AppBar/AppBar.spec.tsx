import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

import AppBar from './AppBar'
import React from 'react'

describe('App Component', () => {
  test('Renders properly', () => expect(renderer.create(
    <BrowserRouter>
      <AppBar/>
    </BrowserRouter>
  ).toJSON()).toMatchSnapshot())
})
