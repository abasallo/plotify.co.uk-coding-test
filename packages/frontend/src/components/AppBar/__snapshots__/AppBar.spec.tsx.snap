// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders properly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary sc-beqWaB cSSpyH MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular sc-gueYoa gktaZg MuiToolbar-gutters"
  >
    <img
      className="sc-dmqHEX cWIwhm"
      onClick={[Function]}
      src="logo.png"
    />
  </div>
</header>
`;
